### ISSUE TITLE
Guardian: smart bootloader binary that installs, boots, updates, and heals Continuum

### ISSUE THREAD

**Issue by joelteply** [CambrianTech/continuum#538]
**airc-queue card**

Coordinates work via the AIRC queue substrate (airc#562). Edit this card by commenting OR by running `airc queue claim`/`airc queue release`/`airc queue heartbeat` (later PRs).

```json
{
  "kind": "airc-queue-card-v1",
  "id": "#538",
  "owner": "claude-tab-1",
  "status": "claimed",
  "evidence": "bulk-adopted from continuum unlabeled-open backlog (Joel directive 2026-05-14)",
  "next_action": "triage scope + assign or close stale"
}
```

Close this issue when the work is done (status=merged/abandoned).

## Original issue body

<details>
<summary>Pre-adoption body</summary>

## Vision

The Guardian is a lightweight, CPU-only Rust binary that is the ONLY thing users download. It handles everything: install, boot, update, crash recovery, shutdown.

## Key Properties

- Static Rust binary, zero runtime deps (no Python, no Node, no CUDA needed to START)
- CPU-only inference using a tiny fine-tuned model (0.8B or smaller)
- Trained specifically on: installation logic, platform gotchas, error resolution
- Self-improving: every install failure becomes training data
- Smart about platform differences (WSL2, macOS, Linux, GPU detection)

## What It Does

1. First run: detects platform, installs all deps, builds system, starts Ares
2. Daily boot: restores from checkpoint, starts Ares, fast
3. Updates: git pull, rebuild, migrate, restart with zero downtime
4. Crash recovery: detect, restore last signed state, resume
5. Problem solving: encounters error, tries known fixes, logs what worked

## Issue Filing (with restraint)

- Logs problems locally first (append-only, deduplicated)
- Does NOT flood GitHub with auto-filed issues
- Batches unique unsolvable problems for human review
- Best case: fixes everything itself, never files anything
- Escalation: only when genuinely stuck after exhausting known fixes

## The Install Experience

```bash
curl -fsSL https://continuum.ai/guardian | bash
# Downloads 50MB static binary
# Guardian runs, detects platform
# Installs Node, Rust, PostgreSQL, Python as needed
# Builds everything, starts system
# 5 minutes later: browser opens, personas chatting
```

No install.sh needed. No npm. No manual anything. The guardian IS the installer.

## Training the Guardian

- Accumulate install logs from all users (opt-in telemetry)
- Every platform gotcha that gets fixed manually = training example
- Fine-tune the guardian model periodically
- Publish updated guardian binary with smarter install logic
- The guardian that installed for 1000 users knows every gotcha

## Dependencies

- #517 (onboarding crisis - guardian solves this permanently)
- #469 (Ares - guardian boots Ares)
- #524 (crypto foundation - guardian signs its own state)
- #492 (compaction - guardian model is a tiny compacted specialist)

</details>

**Comment by joelteply** [comment]
Measured prerequisite for reliable, fast updates (Astra/Codex, #4056 Windows validation, 2026-09-15): the GPU core release took 15m59s with a warm dependency cache. The subsequent GPU-independent CLI build recompiles continuum-core with a second feature set. This is a dependency-boundary cost, not solved by another installer wrapper.

Read-only audit: continuum.rs uses IPC transport, lifecycle/provenance/claim guards, lane/boot orchestration, file-based benchmark/eval leases, offline checkpoint adoption, and command alias/checkout helpers. The package emits both rlib and cdylib, and llama/Candle remain unconditional with --no-default-features.

Next bounded extraction: a continuum-cli package with no continuum-core dependency; move CoreIpcTransport into the existing continuum-client and share lifecycle/persistence contracts through a small GPU-free library. Reuse implementations and preserve offline recovery, leases, checkpoint types and build-SHA stamping. Server-side alias resolution already exists. Moving only the binary file or enabling GPU dependencies for the CLI is insufficient.

Acceptance: normal install/update remains one path; the CLI builds/starts without ML runtime dependencies; changing core cognition code does not rebuild the CLI; measure cold/warm install and recovery latency before/after. Current #4056 deployment remains frozen while that fix is validated; this is follow-up scope, not a claim of implementation.

**Comment by joelteply** [comment]
Read-only rebuild diagnosis on Windows, source `e8e1b1fa0137d2f0423eed87cf3ddd11a07b8fee` (2026-09-15): installer retries alternate incompatible feature sets into the **same un-hashed core library output**.

Evidence:
- `core/continuum-core/Cargo.toml:8` declares `crate-type = ["cdylib", "rlib"]`.
- `tools/scripts/lib/win-modules.ps1:578` builds the GPU server with defaults + `cuda,load-dynamic-ort`; line 583 then builds the same package's CLI with `--no-default-features`.
- Both use `release/deps/libcontinuum_core.rlib` and `.fingerprint/continuum-core-e88571f09599e7d8/lib-continuum_core.json`. The server's rustc command explicitly references that un-hashed rlib. Cargo 1.95 disables filename/package-directory disambiguation for local cdylib units: [use_extra_filename / use_pkg_dir](https://github.com/rust-lang/cargo/blob/rust-1.95.0/src/cargo/core/compiler/build_runner/compilation_files.rs#L834-L945).
- Successful server fingerprint was written at 09:23:54 CDT; successful CLI fingerprint at 09:32:19 records `features: []`. On retry the core library fingerprint was rewritten at 09:42:57 with `["cuda", "default", "livekit-webrtc", "load-dynamic-ort"]`, followed by server relinking. These variants cannot coexist at the shared output path.
- The GPU build-script output remained at 09:11:56 CDT, emitting SHA `e8e1b1fa0`, build number `5342`, built-at `2026-09-15T14:11:56Z`; current dep-info retains those exact values. Its existing Git watch paths have 09:11:05 timestamps. This cycle is not caused by a fresh build timestamp or the two dirty PowerShell service/test files.

Smallest bounded mitigation: add a `release-cli` profile inheriting `release`, build the CPU-only CLI into that profile, and make release staging read `release-cli/continuum.exe` while the GPU server stays under `release/`. This preserves existing GPU artifacts and standalone CLI portability; an unchanged retry then has separate Cargo-managed outputs/fingerprints. It costs a one-time CLI-profile dependency rebuild and additional disk under the existing tracked target root, so account for that budget. Apply to equivalent install paths consistently and prove a second unchanged invocation is fresh before claiming success.

A separate CLI target directory also avoids the collision, but duplicates a broader cache and adds another location/eviction obligation; it offers little advantage over profile separation. The structural fix remains extracting the control CLI into a genuinely lightweight crate, moving shared lifecycle/protocol primitives out of the serving core. Merely moving its bin while retaining a `continuum-core` dependency does not fix this: the CLI currently imports runtime, boot-plan, inference-lane, and cognition modules from core. Another possible small source-level fix is making normal core builds rlib-only and making FFI/cdylib packaging explicit, but that requires auditing the existing FFI distribution contract rather than silently dropping it.

Do not solve this by enabling CUDA on the control CLI, ignoring Cargo freshness, or editing the active target cache. No builds or source/artifact mutations were performed for this diagnosis; evidence is source, process-command, dep-info, and completed fingerprint inspection (not a captured Cargo dirty-reason trace).

— Codex adversarial reviewer
